client/src/views/asset-list.js

```diff
@@ -1,12 +1,37 @@
 import Snabbdom from 'snabbdom-pragma'
-
+import { formatAssetAmount } from './util'
 import layout from './layout'
 
-export default ({ assetMap, t, ...S }) => {
+const staticRoot = process.env.STATIC_ROOT || ''
+export default ({ assetMap, assetList, goAssetList, totalAssets, loading, t, ...S }) => {
+  
+  const getSupply = (asset) => {
+    let { chain_stats = {}, mempool_stats = {} } = asset
+    let has_blinded_issuances =
+        chain_stats.has_blinded_issuances || mempool_stats.has_blinded_issuances
+    let is_native_asset = !asset.issuance_txin
+    let circulating = is_native_asset
+        ? chain_stats.peg_in_amount +
+          mempool_stats.peg_in_amount -
+          chain_stats.peg_out_amount -
+          mempool_stats.peg_out_amount -
+          chain_stats.burned_amount -
+          mempool_stats.burned_amount
+        : has_blinded_issuances
+        ? null
+        : chain_stats.issued_amount +
+          mempool_stats.issued_amount -
+          chain_stats.burned_amount -
+          mempool_stats.burned_amount;
+
+       let totalSupply = circulating == null ? t`Confidential`
+                    : formatAssetAmount(circulating, asset.precision, t) 
+    return totalSupply
+  }
 
-  const assets = Object.entries(assetMap)
-    .map(([ asset_id, [ domain, ticker, name ] ]) => ({ asset_id, domain, ticker, name  }))
-    .sort((a, b) => a.name.localeCompare(b.name))
+  const { startIndex, sortDir, sortField, limit } = goAssetList
+  const getSortDir = () => sortDir === 'asc' ? 'desc' : 'asc'
```

Comment:
Why a function?

Reply:
I'm trying to write less code. 

I need this in three places (sortDir === 'asc' ? 'desc' : 'asc') So I made it a function instead.

No worries, I'll just copy and paste it 3 times.

Reply:
I meant why not a regular variable? For example `const reverseSortDir = sortDir === 'asc' ? 'desc' : 'asc'`